src/Harald.WebApi/Infrastructure/Messaging/KafkaConsumerFactory.cs

```diff
@@ -17,15 +16,20 @@ public KafkaConsumerFactory(KafkaConfiguration configuration)
             _configuration = configuration;
         }
 
-        public Consumer<string, string> Create()
+        public IConsumer<string, string> Create()
         {
-            var config = _configuration.AsEnumerable().ToArray();
-            
-            return new Consumer<string, string>(
-                config: config,
-                keyDeserializer: new StringDeserializer(Encoding.UTF8),
-                valueDeserializer: new StringDeserializer(Encoding.UTF8)
-            );
+            var config = new ConsumerConfig(_configuration.GetConsumerConfiguration());
+            var builder = new ConsumerBuilder<string, string>(config);
+            builder.SetErrorHandler(OnKafkaError);
+            return builder.Build();
+        }
+
+        private void OnKafkaError(IConsumer<string, string> producer, Error error)
+        {
+            if (error.IsFatal)
+                Environment.FailFast($"Fatal error in Kafka producer: {error.Reason}. Shutting down...");
```

Comment:
How does FailFast behave inside Docker Container? 
Microsoft documentation tells that it logs to "Windows Application Event Log", but where would this info be present when inside container?

> Immediately terminates a process after writing a message to the Windows Application event log

Reply:
Unsure. It is also present in the already merged equivalent PRs for Capability-service and K8s-janitor. Good catch!